core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -141,6 +148,89 @@ public String getDetails() {
         return this.details;
     }
 
+    /**
+     * The allocation delay value associated with the index (defaulting to node settings if not set).
+     */
+    public long getAllocationDelayTimeoutSetting(Settings settings, Settings indexSettings) {
+        if (reason != Reason.NODE_LEFT) {
+            return 0;
+        }
+        TimeValue delayTimeout = indexSettings.getAsTime(DELAYED_NODE_LEFT_TIMEOUT, settings.getAsTime(DELAYED_NODE_LEFT_TIMEOUT, DEFAULT_DELAYED_NODE_LEFT_TIMEOUT));
+        return Math.max(0l, delayTimeout.millis());
+    }
+
+    /**
+     * The delay in millis when delaying assigning the shard need to expire in.
```

Comment:
Got confused by this and had to go to the code :) - I think this will be clearer "returns the time in millisecond until the an unassigned shard can be reassigned."